### DISCUSSION TITLE
RS41 Pressure decoding

### DISCUSSION THREAD

**Discussion by eben80** [dl9rdz/rdz_ttgo_sonde#188] [Developers corner]
Hi,
I was wondering what the reason is why we are not decoding the pressure values? 
We just didn't get to it yet or is it because of technical or memory constraints?

**Comment by LukePrior** [comment]
I believe the reason is that @mycarda has added all the RS41 weather decoding for us and they aren't in range of any RS41-SGP for them to test the code with. I do think I remember them saying that the actual code is quite minimal so it should definitely be possible.

**Comment by eben80** [comment]
Aaah OK so the raw data is already here:
```
	    #if 0
			   uint32_t pressureMain = getint24(data, 560, p+27);
			   uint32_t pressureRef1 = getint24(data, 560, p+30);
			   uint32_t pressureRef2 = getint24(data, 560, p+33);
	    #endif
```
Its the function for the calculation with the calibration data that is still missing.

**Reply by dl9rdz** [reply]
Ok, I could add the function for the calculation. Its rather straightforward

**Reply by LukePrior** [reply]
> Ok, I could add the function for the calculation. Its rather straightforward

When uploading to SondeHub it should be to 2 decimal places, not 1 so that we match auto rx.

**Comment by eben80** [comment]
Great thank you!
![image](https://user-images.githubusercontent.com/2545460/134493148-e010a6c8-707f-4a3e-ad79-91abbb99ca57.png)

**Comment by LukePrior** [comment]
You can update the table on the main page now.

**Comment by LukePrior** [comment]
This is awesome, if you feel like adding XDATA to make this feature-complete with auto rx there is a function here (understand if you don't want to, just was told it should be fairly easy to add):

https://github.com/projecthorus/radiosonde_auto_rx/blob/master/demod/mod/rs41_sgm.c#L801

The resulting string is uploaded to SondeHub a hexadecimal string (ascii hex) under field xdata.

**Comment by LukePrior** [comment]
Hey while the data for RS41 pressure looks mostly good we are often a pressure of 0 which I am guessing is what happens when we don't have a valid value.

We also always upload 0 pressure for M10/M20 sondes when we don't even decode it so that behaviour needs to be changed.

**Comment by dl9rdz** [comment]
Ok, default value NAN for pressure was missing, added now.

**Comment by eben80** [comment]
This is awesome...
The data in this flight is provided by the latest version and an auto-rx receiver from elsewhere.
The only gaps you can see in the graphs is where the auto-rx receiver was waiting for calibration frames.
Well done @dl9rdz @mycarda @LukePrior @cixio and others, this has been some amazing progress 👏 
![image](https://user-images.githubusercontent.com/2545460/134647002-30cbb741-2959-4ef9-9d4a-ba5893def5e1.png)

**Comment by smarti2019** [comment]
![grafik](https://user-images.githubusercontent.com/47259711/134765408-22ea69da-6a5b-464c-ad32-677edb97deb1.png)

What do I have to set so that I can also see the temperature, pressure, etc. under Data as shown here?

**Reply by eben80** [reply]
Its's a one line modification:
`  sprintf(ptr + strlen(ptr), "</td></tr><tr><td>QTH: %.6f,%.6f h=%.0fm hs=%.0fkm/h vs=%.1fm/s heading=%.0f&deg; temperature=%.2f&deg;C rH=%.1f&percnt; temprHsensor=%.1f&deg;C pressure=%.2fhPa batteryVoltage=%.2fV</td></tr>\n", s->d.lat, s->d.lon, s->d.alt, (s->d.hs / 1000 * 3600), s->d.vs, s->d.dir, s->d.temperature, s->d.relativeHumidity, s->d.tempRHSensor, s->d.pressure, s->d.batteryVoltage);`

@dl9rdz feel free to include it if you like of course.

Additional question to you @dl9rdz about displaying the degrees character correctly on the OLED. Is it possible? Currently its just garbled.
![image](https://user-images.githubusercontent.com/2545460/134803132-c223ecc3-7e55-4905-8cf0-b2fb8f311b59.png)

**Comment by dl9rdz** [comment]
This is not (yet) part of the main version of the software, its a private modification of eben80

**Comment by dl9rdz** [comment]
Degree character on OLED: The screens file needs to be latin1 encoded.(Actually, for other reasons, I was just about to change the file encoding to utf-8, so this problem will likely disappear in the future :-))

In your screens1.txt, the latin1 char 0xB0 was replaced with the utf8 code point U+FFFD (0xEF 0xBF 0xBD).
If you modified the screen file with http://rdzsonde.local/edit.html?file=screens1.txt, then likely your browser did some conversion to utf8.
If you edited the file locally and used pio run --target=uploadfs or Arduino sketch data upload, then your editor messed up the file.

It *should* be correct in the full image.

If you want to modify the file via the web interface, you can use this method instead of the edit.html thing:
```
cd RX_FSK/data
python3 ../../scripts/ttgoconfig put file screens1.txt
```

**Reply by eben80** [reply]
Thanks for that explanation :)
